tron/serialize/runstate/statemanager.py

```diff
@@ -161,7 +161,11 @@ def restore(self, job_names, skip_validation=False):
                 for job_name, job_state in jobs.items()
             }
             for result in concurrent.futures.as_completed(results):
-                jobs[results[result]]["runs"] = result.result()
+                try:
+                    jobs[results[result]]["runs"] = result.result()
+                except Exception as e:
+                    log.error(f"Failed to restore runs for {results[result]}: {e}")
+                    raise SystemExit(f"Error: {e}")
```

Comment:
sys.exit() would be a little clearer here :)

Reply:
that said, i think this means that restore_runs_for_jobs() now has some dead code? i.e., the code for processing data that's just an empty dictionary

Reply:
or is that still possible for jobs that we actually have no saved state for at all? which i guess might be jobs that were somehow created without ever saving data back to dynamo?